### PR TITLE
Add a module for data quality monitoring of a SageMaker Endpoint.

### PR DESCRIPTION
**Pull request by clokep**
## Describe your changes

Adds a new module (`sagemaker-model-monitoring-module`) for monitoring data quality of a SageMaker Endpoint.

Take a look at the added README on how to test this. Unfortunately the out-of-the-box SageMaker Pipelines cannot be used as they don't have the proper steps to output the required data for monitoring.

## Issue ticket number and link

N/A

## Checklist before requesting a review

- [x] I updated `CHANGELOG.MD` with a description of my changes
- [x] If the change was to a module, I ran the code validation script (`scripts/validate.sh`)
- [x] If the change was to a module, I have added thorough tests
- [x] If the change was to a module, I have added/updated the module's README.md
- [x] If a module was added, I added a reference to the module to the repository's README.md
- [x] I verified that my code deploys successfully using `seedfarmer apply`

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

**Comment by malachi-constant** [comment]
### AWS CodeBuild CI Report

* CodeBuild project: AIOpsModuleTestsInfrastruct-doMLXEYsnmxr
* Commit ID: 30e511c79768d15a74c77a04eb6a858daf4f2c09
* Result: SUCCEEDED
* [Build Logs](https://ythfngxmh6.execute-api.us-west-2.amazonaws.com/Prod/buildlogs?key=eb80df26-cb94-4218-b434-f159430b7827%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*

**Comment by malachi-constant** [comment]
### AWS CodeBuild CI Report

* CodeBuild project: AIOpsModuleTestsInfrastruct-doMLXEYsnmxr
* Commit ID: 9cb6da333a695118d070fadf542e7a9a2ec7f122
* Result: SUCCEEDED
* [Build Logs](https://ythfngxmh6.execute-api.us-west-2.amazonaws.com/Prod/buildlogs?key=fc00a92f-55dc-4ed1-94b8-30164a131e05%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*

File: modules/sagemaker/sagemaker-model-monitoring/requirements-dev.in [hunk 1]
```diff
@@ -0,0 +1,12 @@
+awscli
```
Comment (kukushking) [inline modules/sagemaker/sagemaker-model-monitoring/requirements-dev.in:1]: Is this duplicate of `requirements-dev.in`/`requirements-dev.txt` in the root of the repo?

File: modules/sagemaker/sagemaker-model-monitoring/sagemaker_model_monitoring/settings.py [hunk 2]
```diff
@@ -0,0 +1,87 @@
...
+    sagemaker_project_name: Optional[str] = Field(default=None)
+
+    # Data quality monitoring options.
+    data_quality_checkstep_output_prefix: str = Field(default="")
+    data_quality_output_prefix: str = Field(default="")
+    data_quality_instance_count: int = Field(default=1, ge=1)
+    data_quality_instance_type: str = Field(default="ml.m5.large")
+    data_quality_instance_volume_size_in_gb: int = Field(default=20, ge=1)
+    data_quality_max_runtime_in_seconds: int = Field(default=3600, ge=1)
+    data_quality_schedule_expression: str = Field(default="cron(0 * ? * * *)")
```
Comment (kukushking) [inline modules/sagemaker/sagemaker-model-monitoring/sagemaker_model_monitoring/settings.py:47]: nit - can we add custom tags here, too?
```
    tags: Optional[Dict[str, str]] = Field(default=None)
```

File: modules/sagemaker/sagemaker-model-monitoring/requirements-dev.in (hunk 1 above)
Comment (clokep) [reply modules/sagemaker/sagemaker-model-monitoring/requirements-dev.in:1]: It was copied from `modules/sagemaker/sagemaker-endpoint/`, I didn't fully audit what was in it.

**Comment by malachi-constant** [comment]
### AWS CodeBuild CI Report

* CodeBuild project: AIOpsModuleTestsInfrastruct-doMLXEYsnmxr
* Commit ID: 2daae6963af5cc608d699f59eafc573ad1fa5518
* Result: SUCCEEDED
* [Build Logs](https://ythfngxmh6.execute-api.us-west-2.amazonaws.com/Prod/buildlogs?key=d7d58877-d0fd-4f09-bb9d-7c415c274c37%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*

**Review by kukushking — APPROVED** [review]
Looks great, thank you!

File: modules/sagemaker/sagemaker-model-monitoring/requirements-dev.in (hunk 1 above)
Comment (LeonLuttenberger) [reply modules/sagemaker/sagemaker-model-monitoring/requirements-dev.in:1]: I don't think we need to have `requirements-dev.txt` in our module folders. We ignore it during module installation, and also during our static checks. So I think we should delete these two files.

File: modules/sagemaker/sagemaker-model-monitoring/sagemaker_model_monitoring/stack.py [hunk 3]
```diff
@@ -0,0 +1,149 @@
...
+                iam.ManagedPolicy.from_aws_managed_policy_name("AmazonSageMakerFullAccess"),
+            ],
+        )
+
+        NagSuppressions.add_resource_suppressions(
+            model_monitor_role,
+            suppressions=[
+                {
+                    "id": "AwsSolutions-IAM4",
+                    # TODO
```
Comment (LeonLuttenberger) [inline modules/sagemaker/sagemaker-model-monitoring/sagemaker_model_monitoring/stack.py:121]: what's this TODO for? (same goes for line 97)

**Review by LeonLuttenberger — APPROVED** [review]
Great work! I left a couple of minor nitpicky comments.

File: modules/sagemaker/sagemaker-model-monitoring/sagemaker_model_monitoring/stack.py (hunk 3 above)
Comment (clokep) [reply modules/sagemaker/sagemaker-model-monitoring/sagemaker_model_monitoring/stack.py:121]: I believe a combination of:

1. Is there a better way to "fix" the nags.
2. The error strings I think don't make sense, I'll update those.

**Comment by malachi-constant** [comment]
### AWS CodeBuild CI Report

* CodeBuild project: AIOpsModuleTestsInfrastruct-doMLXEYsnmxr
* Commit ID: d30c2306ed3ef1f7713dae4047090a656633fbc1
* Result: SUCCEEDED
* [Build Logs](https://ythfngxmh6.execute-api.us-west-2.amazonaws.com/Prod/buildlogs?key=ef46b9d0-d35b-4757-b466-93c74c35abed%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*

**Review by LeonLuttenberger — APPROVED** [review]

File: modules/sagemaker/sagemaker-model-monitoring/sagemaker_model_monitoring/stack.py (hunk 3 above)
Comment (clokep) [reply modules/sagemaker/sagemaker-model-monitoring/sagemaker_model_monitoring/stack.py]: I ended up improving the TODO comments, but I suspect we could reduce the wildcards here.

**Pull request merged**